core/src/raw/http_util/client.rs

```diff
@@ -105,21 +104,17 @@ impl HttpClient {
             req_builder = req_builder.version(parts.version);
         }
 
-        req_builder = match body {
-            AsyncBody::Empty => req_builder.body(reqwest::Body::from("")),
-            AsyncBody::Bytes(bs) => req_builder.body(reqwest::Body::from(bs)),
-            AsyncBody::Stream(s) => {
-                #[cfg(not(target_arch = "wasm32"))]
-                {
-                    req_builder.body(reqwest::Body::wrap_stream(s))
-                }
-                #[cfg(target_arch = "wasm32")]
-                {
-                    let bs = crate::raw::oio::StreamExt::collect(s).await?;
-                    req_builder.body(reqwest::Body::from(bs))
-                }
+        // Don't set body if body is empty.
+        if !body.is_empty() {
+            #[cfg(not(target_arch = "wasm32"))]
+            {
+                req_builder = req_builder.body(reqwest::Body::wrap_stream(body))
```

Comment:
```
 pub fn wrap_stream<S>(stream: S) -> Body
    where
        S: futures_core::stream::TryStream + Send + Sync + 'static,
        S::Error: Into<Box<dyn std::error::Error + Send + Sync>>,
        Bytes: From<S::Ok>,
    {
        Body::stream(stream)
    }
```


Where did Buffer implement  S? I can't figure out the place.

Reply:
Buffer implements `Stream` in https://github.com/apache/opendal/pull/4458. And all `Stream<Result<T>>` implements `TryStream`.